src/modules/admin/components/Tokens/ColonyTokenEditDialog.tsx

```diff
@@ -10,38 +10,55 @@ interface Props {
   cancel: () => void;
   close: () => void;
   nativeTokenAddress: Address;
-  selectedTokens: Address[];
 }
 
 const ColonyTokenEditDialog = ({
   colonyAddress,
   nativeTokenAddress,
-  selectedTokens = [],
   cancel,
   close,
 }: Props) => {
   const [setColonyTokensMutation] = useSetColonyTokensMutation();
 
-  const setColonyTokens = useCallback(
-    ({ tokens }) => {
-      setColonyTokensMutation({
-        variables: { input: { colonyAddress, tokenAddresses: tokens } },
+  const { data } = useColonyTokensQuery({
+    variables: { address: colonyAddress },
+  });
+
+  const colonyTokens = (data && data.colony.tokens) || [];
+
+  const addToken = useCallback(
```

Comment:
I actually thought the changes will only be applied if the user presses the submit button on the dialog. It seems they are applied directly which pretty much renders the cancel button nonsensical?